packages/cli-hydrogen/src/cli/services/deploy/config.ts

```diff
@@ -0,0 +1,39 @@
+import {DeployConfig, ReqDeployConfig} from './types.js'
+import {gitInit} from '../../prompts/git-init.js'
+import {git, error} from '@shopify/cli-kit'
+
+export const validateProject = async (config: DeployConfig) => {
+  await git.ensurePresentOrAbort()
+  try {
+    await git.ensureInsideGitDirectory(config.path)
+  } catch (err: unknown) {
+    if ((err as error.Abort)?.message !== git.OutsideGitDirectoryError().message) throw error
+    await initializeGit(config)
+  }
+}
+
+export const initializeGit = async (config: DeployConfig) => {
+  const shouldGitInit = await gitInit()
+
+  if (!shouldGitInit) throw new error.AbortSilent()
+
+  await git.initializeRepository(config.path)
+  git.createGitIgnore(config.path, git.GitIgnoreTemplate.Hydrogen)
+  await git.commitAll('Initial commit generated by deploy-hydrogen', {directory: config.path})
```

Comment:
I don't think we need to be so specific about which script did it, I think "Hydrogen", the entity as a whole, is the active party here.

```suggestion
  await git.commitAll('Initial commit generated by Hydrogen', {directory: config.path})
```